### PR TITLE
Cody: Fix auth view

### PR DESCRIPTION
**Pull request by abeatrix**
Updated so that a user is considered `authed` only when`authStatus.hasVerifiedEmail === true`: `await vscode.commands.executeCommand('setContext', 'cody.activated', authStatus.hasVerifiedEmail)` (When `cody.activated` is set to true, it will activate all the commands for the users on the right conditions). 

While I was working on this, I discovered another bug - see this loom for the UI bug where unauthed users can see visit the `chat` view and `history` view in our current version: 
https://www.loom.com/share/002035051d0341a6833fa8c79db0036e

## Test plan

See this loom for the fixed version: https://www.loom.com/share/1b013db11a9841b8a42689e922818d8d

### CODE REVIEW

**Comment by sourcegraph-buildkite** [comment]
## Bundle size report 📦

| Initial size | Total size | Async size | Modules |
| --- | --- | --- | --- |
| 0.00% (0.00 kb) | 0.00% (+0.33 kb) | 0.00% (+0.33 kb) | 0.00% (0) |

Look at the [Statoscope report](https://console.cloud.google.com/storage/browser/_details/sourcegraph_reports/statoscope-reports/bee/verified-email-cody-vscode/compare-report.html;tab=live_object) for a full comparison between the commits 72b2dcc and 7ee9afc or [learn more](https://docs.sourcegraph.com/dev/how-to/testing#bundlesize).


<details>
<summary>Open explanation</summary>

- `Initial size` is the size of the initial bundle (the one that is loaded when you open the page)
- `Total size` is the size of the initial bundle + all the async loaded chunks
- `Async size` is the size of all the async loaded chunks
- `Modules` is the number of modules in the initial bundle
</details>

**Review by dominiccooney — COMMENTED** [review]
It is time to add some integration tests for this. As you mentioned, while fixing one bug you discovered other ways that it is broken. And we are pushing a boolean parameter through the untyped executeCommand API, which is easy to break in future.

I believe we can flip vscode into a mode where it uses iframes for webviews, and then attach puppeteer to the devtools debugging port to start running Puppeteer assertions about what is in the web views.

**Comment by mrnugget** [comment]
> Updated so that a user is considered `authed` only when`authStatus.hasVerifiedEmail === true`

To be clear: that's only if `site.requiresVerifiedEmailForCody`, otherwise we don't need to check -- right?

File: client/cody-shared/src/sourcegraph-api/graphql/client.ts [hunk 1]
```diff
@@ -26,6 +26,7 @@ interface CurrentUserIdResponse {
 
 interface CurrentUserIdVerificationStatusResponse {
     currentUser: { id: string; hasVerifiedEmail: boolean } | null
+    site: { requiresVerifiedEmailForCody: boolean }
```
Comment (mrnugget) [inline client/cody-shared/src/sourcegraph-api/graphql/client.ts:29]: This is unused, right?

File: client/cody/src/chat/ChatViewProvider.ts [hunk 2]
```diff
@@ -181,8 +181,8 @@ export class ChatViewProvider implements vscode.WebviewViewProvider, vscode.Disp
                     customHeaders: this.config.customHeaders,
                 })
                 // activate when user has valid login
-                await vscode.commands.executeCommand('setContext', 'cody.activated', authStatus)
-                if (authStatus.loggedIn) {
+                await vscode.commands.executeCommand('setContext', 'cody.activated', authStatus.hasVerifiedEmail)
+                if (authStatus.loggedIn && authStatus.hasVerifiedEmail) {
```
Comment (mrnugget) [inline client/cody/src/chat/ChatViewProvider.ts:185]: I think this is wrong here. `authStatus.hasVerifiedEmail` should only affect login status if and only if `site.requiresVerifiedEmailForCody`.

**Review by mrnugget — CHANGES_REQUESTED** [review]
I think the check needs to be changed. Everywhere we do 

```javascript
loggedIn && hasVerifiedEmail
```

it should be

```javascript
loggedIn && (requiresVerifiedEmailForCody ? hasVerifiedEmail : true)
```

In my PR I treated "logged in by doesn't have verified email" as "loggedIn", in your PR that's not the case anymore. I actually think your approach is better, because it's clearer. i.e. you can't login if you don't have a verified email.

But we need to get the check right.

**Comment by mrnugget** [comment]
@abeatrix I rebased my branch after a lot of stuff has changed in `main`, so I think it might be best if we collab on my branch? (If you want!) :)

File: client/cody/src/chat/ChatViewProvider.ts (hunk 2 above)
Comment (thenamankumar) [reply client/cody/src/chat/ChatViewProvider.ts:185]: +1

File: client/cody/src/chat/ChatViewProvider.ts [hunk 3]
```diff
@@ -497,13 +501,16 @@ export class ChatViewProvider implements vscode.WebviewViewProvider, vscode.Disp
     /**
      * Save Login state to webview
      */
-    public async sendLogin(isValid: boolean): Promise<void> {
+    public async sendLogin(authStatus: AuthStatus): Promise<void> {
         this.sendEvent('token', 'Set')
-        await vscode.commands.executeCommand('setContext', 'cody.activated', isValid)
-        if (isValid) {
+        await vscode.commands.executeCommand('setContext', 'cody.activated', authStatus.hasVerifiedEmail)
+        if (authStatus.loggedIn && authStatus.hasVerifiedEmail) {
```
Comment (thenamankumar) [inline client/cody/src/chat/ChatViewProvider.ts:507]: Check for site.requiresVerifiedEmailForCody

**Review by thenamankumar — APPROVED** [review]

**Comment by abeatrix** [comment]
> @abeatrix I rebased my branch after a lot of stuff has changed in `main`, so I think it might be best if we collab on my branch? (If you want!) :)

Yea I missed the cases where we only check that for dotcom so what you suggested sounds good to me!

**Pull request closed without merging**